src/main/java/de/tum/in/www1/artemis/web/rest/TextSubmissionResource.java

```diff
@@ -214,4 +212,86 @@ private void hideDetails(@RequestBody TextSubmission textSubmission) {
                 .collect(Collectors.toList())
         );
     }
+
+    /**
+     * GET  /text-submissions-assessed-by-tutor : get all the textSubmissions assessed by the current tutor.
+     *
+     * @return the ResponseEntity with status 200 (OK) and the list of textSubmissions in body
+     */
+    @GetMapping(value = "/exercises/{exerciseId}/text-submissions-assessed-by-tutor")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    @Transactional(readOnly = true)
+    public ResponseEntity<List<TextSubmission>> getAllTextSubmissionsByTutor(@PathVariable Long exerciseId) {
+        log.debug("REST request to get all TextSubmissions filtered by assessed by tutor");
+        Exercise exercise = exerciseService.findOneLoadParticipations(exerciseId);
+        User user = userService.getUserWithGroupsAndAuthorities();
+
+        if (!hasAtLeastTAPermissions(exercise, user)) return forbidden();
+
+        return ResponseEntity.ok().body(
+
+            // Participations for Exercise
+            participationService.findByExerciseIdWithEagerSubmissions(exerciseId)
+                .stream()
+                .peek(participation -> {
+                    participation.getExercise().setParticipations(null);
+                })
+
+                // Map to Latest Submission
+                .map(Participation::findLatestTextSubmission)
+                .filter(Objects::nonNull)
+
+                // It needs to be submitted to have an assessment
+                .filter(Submission::isSubmitted)
+
+                .filter(textSubmission -> {
+                    Result result = resultRepository.findDistinctBySubmissionId(textSubmission.getId()).orElse(null);
+                    if (result == null) {
+                        return false;
+                    }
+
+                    return result.getAssessor() == user;
+                })
+
+                // Load Result for Submission
+                .peek(textSubmission -> {
+                    Hibernate.initialize(textSubmission.getResult()); // eagerly load the association
+                    if (textSubmission.getResult() != null) {
+                        Hibernate.initialize(textSubmission.getResult().getAssessor());
+                        textSubmission.getResult().getAssessor().setGroups(null);
+                    }
+                })
+
+                // Convert Stream to List to Match Return Type
+                .collect(Collectors.toList())
```

Comment:
I don't like this style because it breaks with how the existing code is structured. Also there are lots of hidden DB calls within the filters and streams. It would be much better to rework the code and try to offload everything into one or two DB calls to make sure we have a good performance also in cases where one tutor corrects hundreds of text submissions.

Reply:
So please rework

Reply:
So I have rewritten all the logic, moving it to the service.

The idea now is we take all the results associated to this exercise and this tutor, and from there we take the submissions. It's the minimum number of calls, and code is easier to read